modules/vault-cluster/main.tf

```diff
@@ -72,6 +111,77 @@ resource "aws_launch_configuration" "launch_configuration" {
   }
 }
 
+# ---------------------------------------------------------------------------------------------------------------------
+# CREATE LAUNCH TEMPLATE TO DEFINE WHAT RUNS ON EACH INSTANCE IN THE ASG
+# ---------------------------------------------------------------------------------------------------------------------
+
+data "aws_ami" "ami" {
+  filter {
+    name   = "image-id"
+    values = ["${var.ami_id}"]
+  }
+}
+
+resource "aws_launch_template" "launch_template" {
+  count         = "${var.asg_launch_mechanism == "launch_template" ? 1 : 0}"
+  name_prefix   = "${var.cluster_name}-"
+  image_id      = "${var.ami_id}"
+  instance_type = "${var.instance_type}"
+  user_data     = "${base64encode(var.user_data)}"
+
+  depends_on = ["aws_iam_instance_profile.instance_profile"]
+
+  iam_instance_profile {
+    name = "${aws_iam_instance_profile.instance_profile.name}"
+  }
+
+  key_name = "${var.ssh_key_name}"
+
+  # Don't use vpc_security_group_ids when network_interfaces includes security_groups.
+  # vpc_security_group_ids = ["${concat(list(aws_security_group.lc_security_group.id), var.additional_security_group_ids)}"]
+
+  placement {
+    tenancy = "${var.tenancy}"
+  }
+  network_interfaces {
+    associate_public_ip_address = "${var.associate_public_ip_address}"
+    delete_on_termination       = true
+    security_groups             = ["${concat(list(aws_security_group.lc_security_group.id), var.additional_security_group_ids)}"]
+  }
+  ebs_optimized = "${var.root_volume_ebs_optimized}"
+  block_device_mappings {
+    device_name = "${data.aws_ami.ami.root_device_name}"
+
+    ebs {
+      encrypted             = "${var.root_volume_ebs_encryption}"
+      volume_type           = "${var.root_volume_type}"
+      volume_size           = "${var.root_volume_size}"
+      delete_on_termination = "${var.root_volume_delete_on_termination}"
+    }
+  }
+  tags = "${var.launch_template_tags}"
+  tag_specifications {
+    # Instance tags are already handled by the autoscaling group
+    resource_type = "volume"
+
+    tags = "${merge(
+      var.volume_extra_tags,
+      map("key", var.cluster_tag_key, "value", var.cluster_name)
+      )
+    }"
+  }
+  # Important note: whenever using a launch configuration with an auto scaling group, you must set
```

Comment:
Is this true with a launch template too?

Reply:
To be honest, I'm not entirely sure that it is (still?) true for launch_configuration.  My code is uses the vault-cluster default (true) and I haven't made any special configuration for the other resources.